go/pkg/client/retries_test.go

```diff
@@ -276,58 +288,71 @@ func TestWriteRetries(t *testing.T) {
 func TestReadRetries(t *testing.T) {
 	t.Parallel()
 	for _, sleep := range []bool{false, true} {
-		sleep := sleep
-		t.Run(fmt.Sprintf("sleep=%t", sleep), func(t *testing.T) {
-			t.Parallel()
-			f := setup(t)
-			defer f.shutDown()
-			if sleep {
-				f.fake.sleepDelay = time.Second
-				client.RPCTimeouts(map[string]time.Duration{"default": 500 * time.Millisecond}).Apply(f.client)
-			}
-
-			blob := []byte("blob")
-			got, err := f.client.ReadBlob(f.ctx, digest.NewFromBlob(blob))
-			if err != nil {
-				t.Errorf("client.ReadBlob(ctx, digest) gave error %s, want nil", err)
-			}
-			if diff := cmp.Diff(blob, got, cmpopts.EquateEmpty()); diff != "" {
-				t.Errorf("client.ReadBlob(ctx, digest) gave diff (-want, +got):\n%s", diff)
-			}
-		})
+		for _, comp := range []bool{false, true} {
```

Comment:
How much longer did it make the test? If it added a few seconds, then suggestion: make compression retry test a separate Test function from this one that includes the sleeping, because these two are really orthogonal. Ditto for files.

Reply:
The non race tests still take < 10s (less than 1s of noticeable increase).

I'm not sure they're orthogonal - I'd argue they _should_ be the same. As much as possible, comp vs noncomp should behave the same way.

Reply:
Sure, if it doesn't make it noticeably longer, this is fine!